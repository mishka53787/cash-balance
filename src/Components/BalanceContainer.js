import React, { useState } from 'react';

import { useSelector, useDispatch } from 'react-redux';
import { deposit, withdraw } from './redux/actions';

const BalanceContainer = () => {
  const [amount, setAmount] = useState('');
  const balance = useSelector((state) => state.balance);
  const dispatch = useDispatch();

  const handleDeposit = () => {
    if (amount) {
      dispatch(deposit(parseFloat(amount)));
      setAmount('');
    }
  };

  const handleWithdraw = () => {
    if (amount) {
      dispatch(withdraw(parseFloat(amount)));
      setAmount('');
    }
  };

  // Rest of the component code
};

export default BalanceContainer;
;

