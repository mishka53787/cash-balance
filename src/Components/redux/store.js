// Store.js
import React from 'react';
import { createStore } from 'redux';
import reducer from './reducer';
import BalanceContainer from '../BalanceContainer'; // Assuming this is your actual component

import { Provider } from 'react-redux';

const store = createStore(reducer);

const AppWithProvider = () => (
  <Provider store={store}>
    <BalanceContainer />
  </Provider>
);

export default AppWithProvider;
