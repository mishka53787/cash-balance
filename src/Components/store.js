import React from 'react';
import { createStore } from 'redux';
import reducer from './redux/reducer';
import BalanceContainer from './BalanceContainer';
import { Provider } from 'react-redux';

const store = createStore(reducer);

const AppWithProvider = () => (
  <Provider store={store}>
    <BalanceContainer />
  </Provider>
);

export default AppWithProvider;

