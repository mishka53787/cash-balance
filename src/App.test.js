import React from 'react';
import { render } from '@testing-library/react';
import BalanceContainer from './Components/BalanceContainer';
import { Provider } from 'react-redux';
import store from './Components/redux/store';

describe('BalanceContainer', () => {
  test('renders component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BalanceContainer />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test('handles button click correctly', () => {
    render(
      <Provider store={store}>
        <BalanceContainer />
      </Provider>
    );
    // Add assertions here for button click functionality
  });
});


// ! wont pass because your App.js file no longer containers the words "learn react"
describe('App', () => {
  test('renders learn react link', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const linkElement = screen.getByRole('link', { name: /learn react/i });
    expect(linkElement).toBeInTheDocument();
  });
});

describe('BalanceContainer', () => {
  test('renders component correctly', () => {
    render(
      <Provider store={store}>
        <BalanceContainer />
      </Provider>
    );
    // Add assertions here
  });

  test('handles button click correctly', () => {
    render(
      <Provider store={store}>
        <BalanceContainer />
      </Provider>
    );
    // Add assertions here
  });
});

// jest.config.js
module.exports = {
  transform: {
    '^.+\\.jsx?$': 'babel-jest',
  },
};
