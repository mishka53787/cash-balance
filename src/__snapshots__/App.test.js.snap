// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceContainer renders component correctly 1`] = `<div />`;
